internal/tsetse/rules/equals_nan_rule.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * @fileoverview Bans `== NaN` in TypeScript code, since it is always false for
+ * any value.
+ */
+
+import * as tsutils from 'tsutils';
+import * as ts from 'typescript';
+
+import {Checker} from '../checker';
+import {AbstractRule} from '../rule';
+
+const FAILURE_STRING = 'x == NaN always returns false; use isNaN(x) instead';
+
+export class Rule extends AbstractRule {
+  readonly ruleName = 'equals-nan';
+  readonly code = 21222;
```

Comment:
hmm, we don't have a system for assigning out these numbers in a unique way. Maybe all the codes should be in one file that we import from, so we don't accidentally collide. Would you mind making that change, even though it's outside your scope?

Reply:
Done. What do we want to do about internal rules? Maybe a different number space? Where/how did we get "allocated" these numbers?